tests/scripts/all.sh

```diff
@@ -3693,6 +3693,69 @@ support_test_cmake_as_package_install () {
     support_test_cmake_out_of_source
 }
 
+component_build_cmake_custom_config_file () {
+    # Make a copy of config file to use for the in-tree test
+    cp "$CONFIG_H" include/mbedtls_config_in_tree_copy.h
+
+    MBEDTLS_ROOT_DIR="$PWD"
+    mkdir "$OUT_OF_SOURCE_DIR"
+    cd "$OUT_OF_SOURCE_DIR"
+
+    # Build once to get the generated files (which need an intact config file)
+    cmake "$MBEDTLS_ROOT_DIR"
+    make
```

Comment:
The scripts that build the generated files use `mbedtls_config.h` instead of respecting `MBEDTLS_CONFIG_FILE` as they want to build with the default configuration, not the custom configuration (custom configs may not contain many options, for example). One of the scripts builds a temporary C file and runs it to get some values of C definitions. This is what is failing in your build.

An intact `mbedtls_config.h` is needed to generate files. For real users this is not a problem as they can leave `mbedtls_config.h` untouched.

However, for the purposes of testing we need to put `#error ...` in `mbedtls_config.h` so we have to build the generated files first, before we modify `mbedtls_config.h`:

```sh
# Build generated files
cmake $MBEDTLS_ROOT_DIR && make
# Setup mbedtls_config etc ...
# Build with the custom config, the -DGEN_FILES=OFF tells cmake not to generate files
cmake -DMBEDTLS_CONFIG_FILE=full_config.h -DGEN_FILES=OFF
make
```

Reply:
I haven't followed everything that's going on here, but in the context of `all.sh`, all components start with the generated files already present in the source tree.